lib/vquic/ngtcp2.c

```diff
@@ -880,9 +882,12 @@ static int ng_perform_getsock(const struct connectdata *conn,
   return ng_getsock((struct connectdata *)conn, socks);
 }
 
-static CURLcode qs_disconnect(struct quicsocket *qs)
+static void qs_disconnect(struct quicsocket *qs)
 {
   int i;
+  if(!qs->conn) /* already closed */
+    return;
+  qs->conn = NULL;
```

Comment:
With this change, is it still needed to set `qs->ssl = NULL` below?

Reply:
Yes, but instead I'll remove the extra `SSL_free()` from `quic_init_ssl` ...